inc/Remote/XmlRpcServer.php

```diff
@@ -67,4 +63,23 @@ protected function call($methodname, $args)
             return new Error($e->getCode(), $e->getMessage());
         }
     }
+
+    /**
+     * Checks the correct content type
+     * 
+     * @param string $contentType
+     * @return bool
+     */
+    protected function checkContentType($contentType){
+        $contentTypeArray = explode(';',$contentType);
+        $typeSize = count($contentTypeArray);
+        
+        for($i =0;$i < $typeSize; $i++){
+            $typeToCheck = strtolower($contentTypeArray[$i]);
```

Comment:
the [specification](http://xmlrpc.com/spec.md) only allows for `text/xml`, I don't think adding more leniency to accept invalid options improves the quality of DW

Reply:
The problem with this is that we don't know the charset of the request, if the client send some "invalid" characters for the charset that the server can handle we will "never" know.
This prevent the server from converting the charset and accepting the request.
Nowadays it's very common that libraries or softwares automatically input the charset in this header and even if the request is XML the server will refuse. [Content-Type on MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type)
The Java sample lib that dokuwiki isn't working anymore on the latest release of dokuwiki because it puts the charset in this header. [DokuWiki Samples](https://www.dokuwiki.org/devel:xmlrpc:clients#sample_java_client), [Java Lib from Sample](https://github.com/gturri/dokujclient)
Maybe this specification is old enough (1999/2003) that this was not common.